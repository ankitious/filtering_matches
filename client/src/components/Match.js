import React from 'react';
import {div} from 'styled-components';

const Match = ({display_name : name,
                   job_title : title,
                   city : {name : cityName},
                   age,
                   main_photo : photo,
                   height_in_cm : height,
                   compatibility_score : compatibility,
                   contacts_exchanged : contacts,
                   religion
               }) =>
 <div style={{ 'display' :'flex',border : '1px solid lightgray'}}>
     <div
         style={{ 'display' : 'flex',
                 'flexDirection' : 'row',
                 'padding': '10px',
                 'width': '25%',
                }}
     >
         <img
             style={{'borderRadius' : '100px',
                     'verticalAlign': 'text-top',
                      'width' : '70px',
                       'height' : '70px',
                   }}
             alt={'profile avatar'}
             src={photo} />

         <div style={{ 'padding' : '0px 10px'
                    }}
         >
            <div>{name}</div>
             <div>{title}</div>
             <div>{age}</div>
             <div>{cityName}</div>
         </div>
     </div>
     <div style={{'paddingTop': '10px', 'paddingLeft' : '5px'}}>
         <div>Height : {height} cm</div>
         <div>Compatibility Score : {compatibility}</div>
         <div>Contacts Exchanged : {contacts}</div>
         <div>Religion : {religion}</div>
     </div>
 </div>;



export default Match;
