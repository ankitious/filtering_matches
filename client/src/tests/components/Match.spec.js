import React from 'react';
import renderer from 'react-test-renderer';
import Match from '../../components/Match';

it('renders correctly', () => {
    const match = {
        display_name: 'Martha',
        job_title: 'Lawyer',
        city: {
            name: 'New York',
        },
        age: 28,
        main_photo: 'http://thecatapi.com/api/images/get?format=src&type=gif',
        height_in_cm: '173cm',
        compatibility_score: 0.75,
        contacts_exchanged: 3,
        religion: 'Atheist'
    };
    const tree = renderer
        .create(<Match {...match} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
