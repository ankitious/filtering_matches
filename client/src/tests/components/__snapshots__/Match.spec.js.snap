// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "border": "1px solid lightgray",
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <img
      alt="profile avatar"
      src="http://thecatapi.com/api/images/get?format=src&type=gif"
      style={
        Object {
          "borderRadius": "100px",
          "height": "70px",
          "verticalAlign": "text-top",
          "width": "70px",
        }
      }
    />
    <div
      style={
        Object {
          "padding": "0px 10px",
        }
      }
    >
      <div>
        Martha
      </div>
      <div>
        Lawyer
      </div>
      <div>
        28
      </div>
      <div>
        New York
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "paddingLeft": "5px",
        "paddingTop": "10px",
      }
    }
  >
    <div>
      Height : 
      173cm
       cm
    </div>
    <div>
      Compatibility Score : 
      0.75
    </div>
    <div>
      Contacts Exchanged : 
      3
    </div>
    <div>
      Religion : 
      Atheist
    </div>
  </div>
</div>
`;
